# Graded assignment 1

Mark Klik & Misja Mikkers

Juanita Marin u250614


# Packages

We need the package `tidyverse`

In [72]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [73]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3         V4        V5        V6        V7       
1 0.8281725 0.9908882 0.25234963 0.7501010 0.7577850 0.9257657 0.1094230
2 0.6554379 0.6502601 0.04260713 0.9330487 0.7506476 0.2390588 0.1474123
  V8        V9         V10       V11        V12       V13       V14       
1 0.9866000 0.31304743 0.1229996 0.32831888 0.6859215 0.3925502 0.47296299
2 0.1469469 0.03453798 0.5179528 0.05019006 0.2700197 0.5642389 0.08337442
  V15       V16       V17       V18         V19       V20      
1 0.6134780 0.2398855 0.1301922 0.176041719 0.7262256 0.1794496
2 0.6081188 0.5881740 0.1895514 0.005803361 0.4767738 0.8708143

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [93]:
df2 <- df1 %>% gather(Key, Value, V1:V20)

head(df2)



Key Value     
1 V1  0.82817254
2 V1  0.65543788
3 V2  0.99088817
4 V2  0.65026007
5 V3  0.25234963
6 V3  0.04260713

## Assignment 2

We will generate some data:

In [96]:
df3 <- as.data.frame(read.table(header=TRUE, text=
        'id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080'))



head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [97]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [103]:
df5 <- rbind(df3, df5)

dfComb[1:4, ]


df5

ERROR: Error in rbind(df3, df5): object 'df5' not found


End of notebook